satpy/readers/viirs_l2.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2011-2019 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Interface to VIIRS L2 files."""
+
+import logging
+from datetime import datetime
+
+from satpy.readers.netcdf_utils import NetCDF4FileHandler
+
+LOG = logging.getLogger(__name__)
+
+
+class VIIRSCloudMaskFileHandler(NetCDF4FileHandler):
+    """VIIRS L2 Cloud Mask reader."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Initialize the file handler."""
+        super().__init__(filename, filename_info, filetype_info, cache_handle=True)
+
+    def _parse_datetime(self, datestr):
+        """Parse datetime."""
+        return datetime.strptime(datestr, "%Y-%m-%dT%H:%M:%SZ")
+
+    @property
+    def start_orbit_number(self):
+        """Get start orbit number."""
+        return int(self['/attr/start_orbit_number'])
+
+    @property
+    def end_orbit_number(self):
+        """Get end orbit number."""
+        return int(self['/attr/end_orbit_number'])
+
+    @property
+    def platform_name(self):
+        """Get platform name."""
+        try:
+            res = self.filename_info['platform_shortname']
+        except KeyError:
+            res = 'Unknown'
+
+        return {
+            'npp': 'Suomi-NPP',
+            'j01': 'NOAA-20',
+            'j02': 'NOAA-21',
+        }.get(res, res)
+
+    @property
+    def sensor_name(self):
+        """Get sensor name."""
+        return self['/attr/instrument_name'].lower()
+
+    def get_shape(self, ds_id, ds_info):
+        """Get shape."""
+        return self.get(ds_id['name'] + '/shape', 1)
+
+    @property
+    def start_time(self):
+        """Get start time."""
+        return self._parse_datetime(self['/attr/time_coverage_start'])
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        return self._parse_datetime(self['/attr/time_coverage_end'])
+
+    def get_metadata(self, dataset_id, ds_info):
+        """Get metadata."""
+        var_path = ds_info['file_key']
+        shape = self.get_shape(dataset_id, ds_info)
+        file_units = self._get_dataset_file_units(ds_info, var_path)
+
+        i = getattr(self[var_path], 'attrs', {})
```

Comment:
Could this variable have a more descriptive name?

Reply:
Done in 732ad5149